practice_app/pages/index.js

```diff
@@ -1,177 +1,735 @@
+    alert("product set has deleted succesfully");
+
+    // sends the input value to getProdcutSet function by using axios.delete
+    axios
+      .delete("./api/getProductCatalog", {
+        data: {
+          type: "productSet",
+          setId: productId,
+        },
+      })
+      .then(function (response) {
+        resultData = response.data;
+        resultStatus = response.status;
+        resultStatusText = response.statusText;
+      });
+  }
+
+  // onclick method for adding a reference image for a product
+  async function onRefImgClick() {
+    // checks whether all the input values are given
+    if (
+      document.getElementById("img-product-title").value === "" ||
+      document.getElementById("img-uri-title").value === ""
+    ) {
+      alert("please fill product id and reference uri");
+      return;
+    }
+
+    // saves the input values
+    productId = document.getElementById("img-product-title").value;
+    referenceImage = document.getElementById("img-uri-title").value;
+
+    // informs the user
+    alert("product image has added succesfully");
+
+    // sends the input value to getProdcutSet function by using axios.post
+    axios
+      .post("./api/getProductCatalog", {
+        uri: referenceImage,
+        type: "refImage",
+        id: productId,
+      })
+      .then(function (response) {
+        resultStatus = response.status;
+        resultStatusText = response.statusText;
+      });
+  }
+
+  return (
+    <div className="container">
+      <Head>
+        <title>Our example API</title>
+        <link rel="icon" href="/favicon.ico" />
+        <link
+          href="https://api.mapbox.com/mapbox-gl-js/v1.10.0/mapbox-gl.css"
+          rel="stylesheet"
+        />
+        <link
+          rel="stylesheet"
+          href="https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v4.5.1/mapbox-gl-geocoder.css"
+          type="text/css"
+        />
+      </Head>
+      <main>
+        <h1>Welcome to our demo website!</h1>
+        {priceConversionText && <p>{priceConversionText}</p>}
+        <button variant="primary" onClick={onConvertButtonClick}>
+          Price Conversion
+        </button>
+        {country && language && currency && <p>{locationGreeting()} </p>}
+        <MapView onMapClick={onMapClick} coordinates={coordinates} />
+
+        <button onClick={getCovidStats}>
+          Click me to get Covid death statistics for the country you have chosen
+          on the map
+        </button>
+        {covidStatsString !== undefined && <p>{covidStatsString}</p>}
+      </main>
+      <Head>
+        <title>Getflix</title>
+      </Head>
+      <h1>Are you vendor?</h1>
+      <h2 class="vendor">Do you want to create your own product set?</h2>
+      <form>
+        <label>
+          SetID:
+          <input
+            type="text"
+            placeholder="Add set id"
+            id="product-set-title"
+            required
+          />
+        </label>
+      </form>
```

Comment:
The HTML `<form>` tag is used to wrap a whole form, not a form element. It should wrap the entire form.